### PR TITLE
Integrate DOI assignment

### PR DESCRIPTION
**Pull request by no-reply**
This implements DOI assignment through to the Datacite API layer.

Configuration for Datacite is set up, and the actor is wired into the stack. Feature and unit tests are introduced to ensure that the DOI is assigned as expected.

### CODE REVIEW

File: app/lib/mahonia/datacite_doi_builder.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    def initialize(prefix: Datacite::Configuration.instance.prefix)
+      super
+    end
+
+    ##
+    # @note hints are ignored
+    #
+    # @see IdentifierBuilder#build
+    def build(*)
+      prefix + '/1234'
```
Comment (bess) [inline app/lib/mahonia/datacite_doi_builder.rb:19]: This is confusing to me. Is this a hard coded value?

File: config/datacite.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+test:
+  domains:   '*'
+  login:    'DATACITE.DCE'
+  password: 'mahoniaD0I'
+  prefix:   '10.5072'
+
+production:
+  domains:   '*'
+  login:    'DATACITE.DCE'
+  password: 'mahoniaD0I'
```
Comment (bess) [inline config/datacite.yml:17]: Can we use dotenv to manage these? We shouldn't check passwords into github, right?

File: spec/lib/mahonia/identifier_builder_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      it 'uses the prefix' do
+        expect(builder.build(hint: 'blah')).to start_with "#{prefix}/"
+      end
+    end
+
+    it 'raises an error with no hint' do
+      expect { builder.build }.to raise_error ArgumentError
+    end
+
+    it 'uses the hint exactly, cast to uppercase' do
```
Comment (bess) [inline spec/lib/mahonia/identifier_builder_spec.rb:38]: This doesn't look cast to uppercase to me. What am I missing?

**Review by bess — CHANGES_REQUESTED** [review]
@no-reply This looks great and I'd really like to understand it better, since I think we're going to be dong a lot of this. I flagged a few small things that confused me.

File: app/lib/mahonia/datacite_doi_builder.rb (hunk 1 above)
Comment (no-reply) [reply app/lib/mahonia/datacite_doi_builder.rb:19]: Yes, temporarily. We need to discuss the identifier selection strategy with OHSU.

File: config/datacite.yml (hunk 2 above)
Comment (no-reply) [reply config/datacite.yml:17]: :+1:

File: spec/lib/mahonia/identifier_builder_spec.rb (hunk 3 above)
Comment (no-reply) [reply spec/lib/mahonia/identifier_builder_spec.rb:38]: this text is just cruft from the Epigaea port. I'll remove it.

**Review by bess — APPROVED** [review]

**Pull request merged**